packages/calcite-components/src/components/rating/rating.tsx

```diff
@@ -259,31 +239,26 @@ export class Rating
           <fieldset class="fieldset" disabled={this.disabled}>
             <legend class="visually-hidden">{this.messages.rating}</legend>
             {this.starsMap.map(
-              ({
-                average,
-                checked,
-                focused,
-                fraction,
-                hovered,
-                id,
-                idx,
-                partial,
-                selected,
-                value,
-              }) => {
+              ({ average, checked, fraction, hovered, id, partial, selected, value, tabIndex }) => {
                 return (
                   <label
                     class={{
                       star: true,
-                      focused,
                       selected,
                       hovered,
                       average,
                       partial,
                     }}
+                    data-value={value}
                     htmlFor={id}
+                    onClick={this.handleLabelClick}
+                    onFocus={this.handleLabelFocus}
```

Comment:
just to note:
We could use `onfocusin` and `onfocusout` on the host instead of `onBlur` and `onFocus` on different elements since `onfocusin` and `onfocusout` bubble whereas blur/focus do not bubble.

Reply:
This made me think and we don't need a `blur` event handler here. And tried using `focusin` at `Host` level which showed weird behavior when user is tabbing out of the rating and then tabbing back in using `Shift+Tab` which is not recognized as focus by `focusin` event.